### DISCUSSION TITLE
babel-preset-expo: Cannot find module 'expo/config'

### DISCUSSION THREAD

**Discussion by mskorenkyi** [expo/expo#27256] [Archived: Questions]
Hello.

I got an error while trying to start my web dev server after upgrading `babel-preset-expo` version to `9.8.0` or higher.

Error text:
```
...
ERROR in ./index.web.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
Error: [BABEL]: Cannot find module 'expo/config'
...
```

I don't use `expo` in my project, but I use `babel-preset-expo`. In the [merge request](https://github.com/expo/expo/pull/24779) was made changes where was added `import { ExpoConfig, getConfig, getNameFromConfig, ProjectConfig } from 'expo/config';` which I guess broke my build. 

However, in the `expo/config.js` file, there is only `module.exports = require('@expo/config');`. So probably it's better to import config from `@expo/config` instead of `expo/config` and add `@expo/config` to the dependencies list or at least import config from `@expo/config` not from `expo/config` because I would need to install `expo` and `@expo/config` dependencies to my project.

I probably missed something, and I'm wrong, and there is a quick fix I can't find by myself.

My `package.json` file:
```
{
    "name": "SomeApp",
    "version": "0.0.1",
    "private": true,
    "scripts": {
        "start-web": "webpack serve --open --config ${path_to_my_config}",
        "test": "jest"
    },
    "dependencies": {
        "@babel/core": "*",
        "@react-native/metro-config": "0.74.0",
        "babel-preset-expo": "10.0.1",
        "react": "18.2.0",
        "react-dom": "18.2.0",
        "react-native": "0.72.6",
        "react-native-web": "0.19.9",
        "react-test-renderer": "18.2.0",
        "webpack": "5.90.3",
        "webpack-cli": "5.1.4",
        "webpack-dev-server": "5.0.2",
        ....
    }
}
```

Any idea how to fix it?

**Comment by aliveornot** [comment]
same problem here

**Comment by airowe** [comment]
@mskorenkyi / @aliveornot Did you find a solution for this?

**Reply by aliveornot** [reply]
I found a dirty workaround for my specific case, but it is highly specific to me and doesn't solve the OP's problem.

**Reply by mskorenkyi** [reply]
I decided not to update `babel-preset-expo` to the latest version and leave it at 9.7.0. When it will be fixed, I'll update my repo.

**Comment by airowe** [comment]
I ended up patching babel-preset-expo to get this working (had to install @expo/config package as well).

> diff --git a/node_modules/babel-preset-expo/build/expo-inline-manifest-plugin.js b/node_modules/babel-preset-expo/build/expo-inline-manifest-plugin.js
> index a93efca..ac32b7d 100644
> --- a/node_modules/babel-preset-expo/build/expo-inline-manifest-plugin.js
> +++ b/node_modules/babel-preset-expo/build/expo-inline-manifest-plugin.js
> @@ -1,7 +1,7 @@
>  "use strict";
>  Object.defineProperty(exports, "__esModule", { value: true });
>  exports.expoInlineManifestPlugin = void 0;
> -const config_1 = require("expo/config");
> +const config_1 = require("@expo/config");
>  const common_1 = require("./common");
>  const debug = require('debug')('expo:babel:inline-manifest');
>  // Convert expo value to PWA value
> diff --git a/node_modules/babel-preset-expo/build/expo-router-plugin.js b/node_modules/babel-preset-expo/build/expo-router-plugin.js
> index 9a9e028..cd65f8b 100644
> --- a/node_modules/babel-preset-expo/build/expo-router-plugin.js
> +++ b/node_modules/babel-preset-expo/build/expo-router-plugin.js
> @@ -4,7 +4,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
>  };
>  Object.defineProperty(exports, "__esModule", { value: true });
>  exports.expoRouterBabelPlugin = void 0;
> -const config_1 = require("expo/config");
> +const config_1 = require("@expo/config");
>  const fs_1 = __importDefault(require("fs"));
>  const path_1 = __importDefault(require("path"));
>  const resolve_from_1 = __importDefault(require("resolve-from"));

**Comment by stevenwchien** [comment]
I had the same issue in my project. After I installed `expo` as a direct dependency, that caused this particular error to go away (I didn't need to patch the `expo-inline-manifest-plugin.js` file). Strangely enough, `expo` doesn't seem to be listed as a peer dependency of `babel-preset-expo`, but rather, it's the peer dependency of another project I was using, which is why I tried this.

**Comment by VGerris** [comment]
None of these tricks worked for me, I ended up upgrading Babel to 7.26.0, delete node_modules and yarn.lock. run yarn and then run this in the apps/expo folder :

yarn expo install --fix   

There may be other issues, I suggest to upgrade to recent packages for all dependencies and also check:

https://react-native-community.github.io/upgrade-helper/

In case you got the datepicker issue at upgrade :

yarn add @react-native-community/datetimepicker

**Comment by LovesWorking** [comment]
.npmrc

ignore-scripts=true


worked for me!

**Comment by arzen07** [comment]
"devDependencies": {
    "@types/react": "~19.1.0",
    "typescript": "~5.9.2",
    "babel-preset-expo": "~12.0.0"
...
  },
  
  For me I was missing babel-present-expo in devDependencies.
